pkg/reconciler/pipelinerun/resources/resultrefresolution.go

```diff
@@ -72,6 +72,14 @@ func extractResultRefsForParam(pipelineRunState PipelineRunState, param v1beta1.
 	return nil, nil
 }
 
+func extractResultRefsForWhenExpression(pipelineRunState PipelineRunState, whenExpression v1beta1.WhenExpression) (ResolvedResultRefs, error) {
+	expressions, ok := whenExpression.GetVarSubstitutionExpressionsForWhenExpression()
+	if ok {
+		return extractResultRefs(expressions, pipelineRunState)
+	}
+	return nil, nil
+}
```

Comment:
this is kind of a nit pick but this doesnt seem worth an entire function to me, the code might be clearer if you called GetVarSubstitutionExpressionsForWhenExpression(), what do you think?